Move actor images on episodes and seasons screens

bowlingbeeg opened the pull request:
GHI (If applicable): #

## Description:
I've been working on this for a while now and clearly I'm not a GUI developer :) but thought I'd upload these changes as a starting base.  After you moved the actor images on the pre-play screen based on the plex web client I figured we should probably do the same thing for the seasons and episodes screens.  So this is my attempt at that.  But some of the calculations to figure out where the drop down box should go seemed really arbitrary so I just kind of fiddled until something worked.  Maybe you have a better understanding of how those numbers were picked in the first place.  When seasons are displayed(on the seasons screen or episodes screen) I tried to get the actor image and name/character name to show up on the screen while the seasons hub was selected.  This may not be the best for spacing but I thought it was nice to see those values without having to fully go to the "role" hub.  If other's don't like that I'm not married to the idea and I can work on adding the spacing back.

Seasons screen:
![Screenshot_20240109-114623](https://github.com/pannal/plex-for-kodi/assets/17151296/94b998cc-b5d5-4818-8b37-f2e871ca3d27)

Episodes screen with other seasons available:
![Screenshot_20240109-114640](https://github.com/pannal/plex-for-kodi/assets/17151296/a8881e2b-2def-45b7-92fe-9933e72d4400)

Episodes screen with no other seasons available:
![Screenshot_20240109-114607](https://github.com/pannal/plex-for-kodi/assets/17151296/3e96bd80-d56a-430f-9b29-443b058afdb1)


## Checklist:
- [X] I have based this PR against the develop branch

pannal commented:
Thank you, where does this massive gap come from? 
![image](https://github.com/pannal/plex-for-kodi/assets/1359593/d3c04cf4-17c2-4df7-8006-f460b001633b)

pannal commented:
Also I think removing the "Other Seasons" header is a bad idea to be honest, just to move the actors into view.

pannal commented:
Hmm, thanks for the effort, but I think it's way too cramped now all in all.

bowlingbeeg commented:
> Thank you, where does this massive gap come from? 

I'm not sure where that gap is coming from.  My pics don't show that.

> Also I think removing the "Other Seasons" header is a bad idea to be honest, just to move the actors into view.

Ok, but how is it any different then removing the "Cast" heading?

> Hmm, thanks for the effort, but I think it's way too cramped now all in all.

I'll add back the spacing.

pannal commented:
> Ok, but how is it any different then removing the "Cast" heading?

Cast is obvious, "Other Seasons" isn't _as_ obvious. 

You've also removed the spacers, right?

pannal commented:
> I'm not sure where that gap is coming from. My pics don't show that.

I've just checked out your branch and started PM4K. That's how the episodes view looks for me using your branch :o

bowlingbeeg commented:
> Cast is obvious, "Other Seasons" isn't _as_ obvious.

Ok .. Personally I think they should all have headings just like the Plex app does, I guess I assumed you wanted to head the other direction and remove them all.  It seems like it should be consistent.

> You've also removed the spacers, right?

If you're referring to the line seperatirs then yes.  Again wanted it to look more like the Plex app.

> I've just checked out your branch and started PM4K. That's how the episodes view looks for me using your branch :o

Hmm, I'll have to look more tonight.  Maybe I missed something when I copied it over from my fork.

Commit:
Move actor images on episodes and seasons screens

bowlingbeeg committed:
Merge branch 'develop_kodi21' into develop_kodi21_actor_images_location

bowlingbeeg commented:
ok, I made some updates and added in the spacing.  I also added back the headers for the the "Cast" because I thought it looked strange to have the Seasons header then the cast with no header followed by the related header.  It also matches more the style of the plex app.  Let me know what you think.

pannal commented:
Yep, looks good, thank you!

I'll remove the "Seasons" heading from the show view, though, as it's too cramped, especially when the show has progress, and I think when you visit a show it's obvious that you're seeing its seasons.

bowlingbeeg commented:
One thing I keep struggling with is the order on the episodes screen.  I personally use the other seasons more often than the cast stuff so I have the cast come after other seasons.  But if there is only one season of a show then there is no other seasons row and the cast comes directly after the episodes list.  So part of me wants to make them the same and have the cast come immediately after the episodes.  The plex app and web don't have those things on the episodes list so there's no guidance.  Not sure if you have an opinion either way.

pannal commented:
Hmm I think we're fine the way it is. We're already deviating from the other Plex apps with the "other seasons" stuff.

bowlingbeeg commented:
Ok sounds good
